src/figuresClasses.ts

```diff
@@ -1,19 +1,80 @@
 export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
 
+  getArea(): number;
 }
 
 export class Triangle {
+  shape = 'triangle';
 
+  constructor(
+    public color: string,
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+  ) {
+    if (sideA <= 0
+      || sideB <= 0
+      || sideC <= 0
+    ) {
+      throw new Error('side must be greater than zero');
+    }
+
+    if (sideA >= sideB + sideC
+      || sideB >= sideA + sideC
+      || sideC >= sideA + sideB
+    ) {
+      throw new Error(
+        'side of the triangle must be less than the sum of the other two',
+      );
+    }
+  }
+
+  getArea(): number {
+    const p = (this.sideA + this.sideB + this.sideC) / 2;
```

Comment:
Variable names should be meaningful, that's why `p` is better rename to `semiPerimetr``.